### DISCUSSION TITLE
Send data alongside server change & listen for it

### DISCUSSION THREAD

**Discussion by zax71** [Minestom/Minestom#1862] [Help] [answered]
Hi! I have code to send a player to another server (below), but I would like to bundle more info with the Plugin message. I found [this event](https://javadoc.minestom.net/net/minestom/server/event/player/PlayerPluginMessageEvent.html), but it listens for sending, not receiving of the messages. How could I send extra data with the server switch plugin message and listen for that on the on the other end?

```java
private static void sendToServer(Player player, String server) {
        // https://github.com/Minestom/Minestom/discussions/1414#discussioncomment-3712097
        ByteArrayDataOutput out = ByteStreams.newDataOutput();
        out.writeUTF("Connect");
        out.writeUTF(server);
        player.sendPacket(new PluginMessagePacket("BungeeCord", out.toByteArray()));
        logger.info("Sent " + player.getUsername() + " to " + server);
    }
```

**Comment by Tofaa2** [comment]
You can create your own channel, where you have BungeeCoord etc, and instead of that specify your own custom channel and send over data the same way you send players. Then on velocity's end you can listen for the plugin message event and route it to the other backend server or do whatever you want with it

**Reply by zax71** [reply]
I was planning to just listen to it on just one backend server, can I do that without a velocity plugin? Also, how do I receive the events at the other end as there are no events that seam to fit?

**Reply by mworzala** [reply, marked as answer]
this is the correct answer. You cannot send a plugin message directly from one backend server to another, they only go from backend server to client (through proxy where they are often intercepted). What Tofaa said should work, and you can listen to plugin messages "from a player" (or in this case the proxy) on the other backend server using `PlayerPluginMessageEvent` (it is for receiving, nothing to do with sending)
